pep-0709.rst

```diff
@@ -126,13 +131,36 @@ the need to place these variables in a cell, allowing the comprehension (and all
 other code in the outer function) to access them as normal fast locals instead.
 This provides further performance gains.
 
+Module-level comprehensions are not inlined
+-------------------------------------------
+
 This PEP only proposes to inline comprehensions occurring inside functions,
 where fast-locals (``LOAD_FAST/STORE_FAST``) are used. Module-level
 comprehensions will continue to create and call a function. Module-level
 comprehensions are generally called only once (when the module is imported), so
-optimizing the performance of calling them is unnecessary, and inlining them
-would require separate code paths in the compiler to handle a module global
-namespace dictionary instead of fast-locals.
+optimizing their performance is low priority. Inlining them would require
+separate code paths in the compiler to handle a module global namespace
+dictionary instead of fast-locals, and it would be difficult or impossible to
+avoid breaking semantics, since the comprehension iteration variable itself
+would be a module global which might be referenced inside other functions,
+which might be called within the comprehension.
+
+Some generator expressions may be inlined
+-----------------------------------------
+
+Generator expressions are a more complex case. A returned generator may escape
+the function where the generator expression occurs, or may only be resumed
+later in the function after other side effects may have occurred. In such cases
+it must be a real resumable generator, which must be backed by its own
+function, and can't be inlined.
+
+There are some cases (code like ``tuple(x for x in lst)`` or ``any(x for x in
+lst)``) where we can prove at compile time that the generator doesn't escape.
+The compiler could identify and inline these, eliminating generator overhead.
+The reference implementation does not yet implement inlining of any generator
+expression (and it should be implemented and reviewed as a separate follow-up
+pull request), but acceptance of this PEP also implies future inlining of
+inlinable generator expressions.
```

Comment:
Likewise, aside from a short statement that some generator expressions may be inlined in the future where it can be proved they do not leak or are resumed after possible side effects, this would likely fit better as an explanation and justification for this particular design decision in the Rationale section, keeping only the concrete specification in the normative Specification section.

Reply:
In the end I decided that this whole section can really be shortened to a single sentence in the specification; there isn't enough more to say to justify a section in the Rationale.